src/libmongoc/src/mongoc/mongoc-secure-channel.c

```diff
@@ -695,8 +710,12 @@ mongoc_secure_channel_handshake_step_2 (mongoc_stream_tls_t *tls,
                return true;
             }
 
-            MONGOC_ERROR (
+            MONGOC_LOG_AND_SET_ERROR (
+               error,
+               MONGOC_ERROR_STREAM,
+               MONGOC_ERROR_STREAM_SOCKET,
                "failed to receive handshake, SSL/TLS connection failed");
+
```

Comment:
Optional: If the expectation is that these `mongoc_secure_channel_handshake_step_2` sets `error` when returning false, I suggest adding a new error message to L684. IIUC that indicates a logic error in the caller. So perhaps the message could be something like:

```
MONGOC_LOG_AND_SET_ERROR (error,
                          MONGOC_ERROR_STREAM,
                          MONGOC_ERROR_STREAM_SOCKET,
                          "Invalid state: credentials or context unset");
```

Noting this is optional, since `mongoc_stream_tls_secure_channel_handshake` sets the generic "TLS handshake failed" in the event that the error is unset.